openmc/mgxs/mgxs.py

```diff
@@ -3694,70 +3758,252 @@ def histogram_bins(self):
-                               endpoint=True)
-            filters = [[energy], [energy, energyout, openmc.MuFilter(bins)]]
+                filters.append([[energyout]])
 
         return self._add_angle_filters(filters)
 
     @property
     def rxn_rate_tally(self):
 
         if self._rxn_rate_tally is None:
-            if self.scatter_format == 'legendre':
+
+            if self.formulation == 'simple':
+                if self.scatter_format == 'legendre':
+                    # If using P0 correction subtract scatter-1 from the diagonal
+                    if self.correction == 'P0' and self.legendre_order == 0:
+                        scatter_p0 = self.tallies['{}-0'.format(self.rxn_type)]
+                        scatter_p1 = self.tallies['{}-1'.format(self.rxn_type)]
+                        energy_filter = scatter_p0.find_filter(openmc.EnergyFilter)
+                        energy_filter = copy.deepcopy(energy_filter)
+                        scatter_p1 = scatter_p1.diagonalize_filter(energy_filter)
+                        self._rxn_rate_tally = scatter_p0 - scatter_p1
+
+                    # Extract scattering moment reaction rate Tally
+                    elif self.legendre_order == 0:
+                        tally_key = '{}-{}'.format(self.rxn_type,
+                                                   self.legendre_order)
+                        self._rxn_rate_tally = self.tallies[tally_key]
+                    else:
+                        tally_key = '{}-P{}'.format(self.rxn_type,
+                                                    self.legendre_order)
+                        self._rxn_rate_tally = self.tallies[tally_key]
+                elif self.scatter_format == 'histogram':
+                    # Extract scattering rate distribution tally
+                    self._rxn_rate_tally = self.tallies[self.rxn_type]
+
+                self._rxn_rate_tally.sparse = self.sparse
+
+            else:
+                msg = 'The reaction rate tally is poorly defined' \
+                      ' for the consistent formulation'
+                raise NotImplementedError(msg)
+
+        return self._rxn_rate_tally
+
+    @property
+    def xs_tally(self):
+        if self._xs_tally is None:
+            if self.tallies is None:
+                msg = 'Unable to get xs_tally since tallies have ' \
+                      'not been loaded from a statepoint'
+                raise ValueError(msg)
+
+            # Use super class method
+            if self.formulation == 'simple':
+                self._xs_tally = MGXS.xs_tally.fget(self)
+
+            else:
+                # Compute groupwise scattering cross section
+                self._xs_tally = self.tallies['scatter'] / \
+                                 self.tallies['flux (tracklength)']
+
+                # Compute scattering probability matrix
+                energyout_bins = [self.energy_groups.get_group_bounds(i)
```

Comment:
Shouldn't there be a special branch around here for histogram scattering?  i.e., the scattering probability won't be in scatter-0, it has to be found from all of the histogram bins (see my `mgxs_library.py` convert_scatter_format for a guide